src/freenet/clients/http/WelcomeToadlet.java

```diff
@@ -91,10 +91,11 @@ private void addCategoryToList(BookmarkCategory cat, HTMLNode list, boolean noAc
                     // extract the key type
                     String keyType = initialKey.substring(1,3);
                     String key = '/' + initialKey + (initialKey.endsWith("/") ? "" : "/");
+					Matcher match = null;
```

Comment:
Seeing that `match` is not used anywhere outside the two branches within the `switch` statement, I believe “fix scope issue” and this change are diametrally opposed. 😀

Reply:
Yes, but the switch branches share the scope, so that’s how it has to be to avoid inventing new names for the different branches.